Extension/src/LanguageServer/extension.ts

```diff
@@ -1162,3 +1162,39 @@ export function UpdateInsidersAccess(): void {
         void vscode.commands.executeCommand("workbench.extensions.installExtension", "ms-vscode.cpptools", { installPreReleaseVersion: true }).then(undefined, logAndReturn.undefined);
     }
 }
+
+export async function PreReleaseCheck(): Promise<void> {
+    // First we need to make sure the user isn't already on a pre-release version.
+    if (!util.packageJson.__metadata?.preRelease === true) {
+        // Get the info on the latest version from the marketplace to check if there is a pre-release version available.
+        const response = await fetch('https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery', {
```

Comment:
Shouldn't we opt people into the prerelease train, even if there isn't one currently available? 
  https://github.com/microsoft/vscode/issues/139528

Reply:
I think it makes more sense to wait until one is available or else it looks like we didn't do anything.

Reply:
I lean towards what Bob said, that we want to see if there's anything to offer before showing a prompt. But happy to change it if we decide otherwise.